### PR TITLE
feat: improve Gemini in chat & UI improvements

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **New Features**
  * Native Gemini streaming with model-aware reasoning (separate reasoning vs. answer segments), tool-call support, Assistant Settings sheet (model selector + "Show reasoning" toggle), modular chat UI, improved input area with voice controls, ambient streaming glow, and refined message bubbles (copy action, timestamps).

* **Bug Fixes**
  * Safer provider-type handling, more robust tool-argument accumulation, improved streaming stability and edge-case handling, and audio transcription ignoring empty chunks.

* **Documentation**
  * Gemini design notes and AI README updates.

* **Tests**
  * Expanded coverage for Gemini, streaming parsing/utils, UI widgets, and config normalization.

### CODE REVIEW

File: lib/features/ai_chat/ui/widgets/chat_interface/message_bubble.dart [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                  clipBehavior: Clip.none,
+                  children: [
+                    Container(
+                      padding: const EdgeInsets.symmetric(
+                          horizontal: 18, vertical: 14),
+                      decoration: BoxDecoration(
+                        color: isUser
+                            ? theme.colorScheme.primary.withValues(alpha: 0.4)
+                            : theme.colorScheme.surfaceContainerHigh,
+                        borderRadius: BorderRadius.circular(18),
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai_chat/ui/widgets/chat_interface/message_bubble.dart:65]: no that's the latest Flutter way

File: test/features/ai_chat/ui/widgets/chat_interface/thinking_parser_test.dart [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      const input = 'pre```think\nBODY without closing fence';
+      final parsed = parseThinking(input);
+      expect(parsed.visible, 'pre');
+      expect(parsed.thinking, isNotNull);
+      // ignore: unnecessary_null_checks
+      expect(parsed.thinking!.startsWith('BODY'), isTrue);
+    });
+
+    test('enforces max thinking length with truncation notice', () {
+      final long = 'A' * (ThinkingUtils.maxThinkingLength + 50);
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply test/features/ai_chat/ui/widgets/chat_interface/thinking_parser_test.dart:116]: probably outdated, compiles fine

File: docs/implementation_plans/2025-09-08_Gemini_API_client.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+## Implementation Details
+
+### 1. Add Dependencies
+
+```yaml
+# pubspec.yaml
+dependencies:
+  google_generative_ai: ^0.5.0  # Official Google SDK
+```
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2025-09-08_Gemini_API_client.md:63]: that was only a plan that was not followed, only her for documentation purposes

File: docs/implementation_plans/2025-09-08_Gemini_API_client.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    @Default(false) bool includeThoughts,
+  }) = _GeminiThinkingConfig;
+  
+  // Presets for common scenarios
+  static const auto = GeminiThinkingConfig(thinkingBudget: -1);
+  static const disabled = GeminiThinkingConfig(thinkingBudget: 0);
+  static const standard = GeminiThinkingConfig(thinkingBudget: 8192);
+  static const intensive = GeminiThinkingConfig(thinkingBudget: 16384);
+}
+```
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2025-09-08_Gemini_API_client.md:107]: just plan at the time, for documentation purposes

File: lib/features/ai_chat/ui/controllers/chat_session_controller.dart [hunk 5]
```diff
@@ -101,17 +114,217 @@ class ChatSessionController extends _$ChatSessionController {
...
+      }
+      // Ensure global streaming flags are cleared even if we ended on a
+      // thinking segment (no active streaming message to finalize).
+      if (state.isLoading || state.isStreaming) {
+        state = state.copyWith(
+          isLoading: false,
+          isStreaming: false,
+          streamingMessageId: null,
+        );
+      }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai_chat/ui/controllers/chat_session_controller.dart:186]: fixed

File: lib/features/ai_chat/ui/widgets/chat_interface/bubble_corner_action.dart [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        elevation: 2,
+        shape: const CircleBorder(),
+        shadowColor: Colors.black.withValues(alpha: 0.3),
+        child: InkWell(
+          customBorder: const CircleBorder(),
+          onTap: onTap,
+          child: Padding(
+            padding: const EdgeInsets.all(6),
+            child: Icon(icon, size: 16, color: theme.colorScheme.onSurface),
+          ),
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/ai_chat/ui/widgets/chat_interface/bubble_corner_action.dart:31]: nope

File: docs/implementation_plans/2025-09-08_Gemini_API_client.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        // Emit thinking block
+        yield '<thinking>\n${thinkingBuffer.toString()}\n</thinking>\n';
+        thinkingBuffer.clear();
+        inThinking = false;
+      }
+      // Emit regular content
+      yield chunk.text ?? '';
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply docs/implementation_plans/2025-09-08_Gemini_API_client.md:168]: yeah this was only the initial plan, for documentation purposes
